contracts/Transfer.sol

```diff
@@ -128,111 +125,29 @@ contract Transfer {
             bytes32,
             bytes memory,
             bytes memory,
-            Types.Result
+            Types.Result result
         )
     {
-        require(
-            MerkleTreeUtilsLib.verifyLeaf(
-                stateRoot,
-                keccak256(from.state.encode()),
-                _tx.fromIndex,
-                from.witness
-            ),
-            "Transfer: sender does not exist"
-        );
+        bytes32 newRoot = bytes32(0);
+        bytes memory newFromState = "";
 
-        Types.Result result = FraudProofHelpers.validateTxBasic(
+        (newRoot, newFromState, result) = Transition.processSender(
+            stateRoot,
+            _tx.fromIndex,
+            tokenType,
             _tx.amount,
             _tx.fee,
-            from.state
+            from
         );
         if (result != Types.Result.Ok) return (bytes32(0), "", "", result);
-
-        if (from.state.tokenType != tokenType) {
-            return (bytes32(0), "", "", Types.Result.BadFromTokenType);
-        }
-
-        if (to.state.tokenType != tokenType)
-            return (bytes32(0), "", "", Types.Result.BadToTokenType);
```

Comment:
We used to have this dangerous line that we judge the stuff provided by the disputer without validating it.
The Merkle verify is at the 10 lines below, but here we are already trying to slash the coordinator by whatever states provided by the disputer.